lib/Sema/CSGen.cpp

```diff
@@ -1020,8 +1020,39 @@ namespace {
       // `_ = nil`, let's diagnose it here because solver can't
       // attempt any types for it.
       auto *parentExpr = CS.getParentExpr(expr);
-      while (parentExpr && isa<ParenExpr>(parentExpr))
+      bool hasContextualType = bool(CS.getContextualType(expr));
+
+      while (parentExpr) {
+        if (!isa<ParenExpr>(parentExpr))
```

Comment:
Should we skip all `IdentityExpr` nodes a la `getSemanticsProvidingExpr()`?

Reply:
I don't know why anybody would write `try nil` or `await nil` in this kind of position in a way where you would not have contextual information, but it's something to think about.

Reply:
I think it makes sense, other `IdentityExpr`s are `.self`, implicit chain result and `await`, seems like all of them shouldn't matter contextually...

Reply:
✅